# Latihan

*   Nama: Diah Ayu Puspasari
*   Rombel: DE02
*   NIM: 0110223052





In [1]:
# import library
import pandas as pd
import numpy as np
from sklearn.preprocessing import MinMaxScaler

In [2]:
# menghubungkan gdrive dengan colab
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# mengimpoert file dari google drive
path = '/content/drive/MyDrive/datasciece/dataset'

In [4]:
# membaca file csv menggunakan pandas
mydata = pd.read_csv(path + '/data-boston-housing.csv')
mydata

Unnamed: 0     CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD  \
0             0  0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900  1.0   
1             1  0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671  2.0   
2             2  0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671  2.0   
3             3  0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622  3.0   
4             4  0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622  3.0   
..          ...      ...   ...    ...   ...    ...    ...   ...     ...  ...   
501         501  0.06263   0.0  11.93   0.0  0.573  6.593  69.1  2.4786  1.0   
502         502  0.04527   0.0  11.93   0.0  0.573  6.120  76.7  2.2875  1.0   
503         503  0.06076   0.0  11.93   0.0  0.573  6.976  91.0  2.1675  1.0   
504         504  0.10959   0.0  11.93   0.0  0.573  6.794  89.3  2.3889  1.0   
505         505  0.04741   0.0  11.93   0.0  0.573  6.030  80.8  2.5050  1.0   

       TAX  PTRATIO       B  LSTAT  MEDV  
0    296.0     15.3  396.90   4.98  24.0  
1    242.0     17.8  396.90   9.14  21.6  
2    242.0     17.8  392.83   4.03  34.7  
3    222.0     18.7  394.63   2.94  33.4  
4    222.0     18.7  396.90   5.33  36.2  
..     ...      ...     ...    ...   ...  
501  273.0     21.0  391.99   9.67  22.4  
502  273.0     21.0  396.90   9.08  20.6  
503  273.0     21.0  396.90   5.64  23.9  
504  273.0     21.0  393.45   6.48  22.0  
505  273.0     21.0  396.90   7.88  11.9  

[506 rows x 15 columns]

In [5]:
# melihat datatype dan spesfikasi data
mydata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  506 non-null    int64  
 1   CRIM        506 non-null    float64
 2   ZN          506 non-null    float64
 3   INDUS       506 non-null    float64
 4   CHAS        506 non-null    float64
 5   NOX         506 non-null    float64
 6   RM          506 non-null    float64
 7   AGE         506 non-null    float64
 8   DIS         506 non-null    float64
 9   RAD         506 non-null    float64
 10  TAX         506 non-null    float64
 11  PTRATIO     506 non-null    float64
 12  B           506 non-null    float64
 13  LSTAT       506 non-null    float64
 14  MEDV        506 non-null    float64
dtypes: float64(14), int64(1)
memory usage: 59.4 KB


In [6]:
# mencari hasil dari statistika deskriptif kolom float
mydata.describe()

Unnamed: 0        CRIM          ZN       INDUS        CHAS         NOX  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean   252.500000    3.613524   11.363636   11.136779    0.069170    0.554695   
std    146.213884    8.601545   23.322453    6.860353    0.253994    0.115878   
min      0.000000    0.006320    0.000000    0.460000    0.000000    0.385000   
25%    126.250000    0.082045    0.000000    5.190000    0.000000    0.449000   
50%    252.500000    0.256510    0.000000    9.690000    0.000000    0.538000   
75%    378.750000    3.677083   12.500000   18.100000    0.000000    0.624000   
max    505.000000   88.976200  100.000000   27.740000    1.000000    0.871000   

               RM         AGE         DIS         RAD         TAX     PTRATIO  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean     6.284634   68.574901    3.795043    9.549407  408.237154   18.455534   
std      0.702617   28.148861    2.105710    8.707259  168.537116    2.164946   
min      3.561000    2.900000    1.129600    1.000000  187.000000   12.600000   
25%      5.885500   45.025000    2.100175    4.000000  279.000000   17.400000   
50%      6.208500   77.500000    3.207450    5.000000  330.000000   19.050000   
75%      6.623500   94.075000    5.188425   24.000000  666.000000   20.200000   
max      8.780000  100.000000   12.126500   24.000000  711.000000   22.000000   

                B       LSTAT        MEDV  
count  506.000000  506.000000  506.000000  
mean   356.674032   12.653063   22.532806  
std     91.294864    7.141062    9.197104  
min      0.320000    1.730000    5.000000  
25%    375.377500    6.950000   17.025000  
50%    391.440000   11.360000   21.200000  
75%    396.225000   16.955000   25.000000  
max    396.900000   37.970000   50.000000

In [7]:
# mengecek missing values data
mydata.isnull().sum()

Unnamed: 0    0
CRIM          0
ZN            0
INDUS         0
CHAS          0
NOX           0
RM            0
AGE           0
DIS           0
RAD           0
TAX           0
PTRATIO       0
B             0
LSTAT         0
MEDV          0
dtype: int64

## Penjelasan Data

1.   Dataset memiliki 506 baris (entries) dan 15 kolom (columns).
2.   Tidak ada nilai null (semua kolom memiliki 506 non-null values).
3.   Dataset terdiri dari angka bertipe float64 dan int64, yang berarti semua fitur bersifat numerik.


Penjelasan Kolom:
*   Unnamed: 0 → Indeks, tidak memiliki makna analitis dan bisa dihapus.
*   CRIM → Tingkat kejahatan per kapita di kota (semakin tinggi, semakin rendah harga rumah)
*   ZN → Persentase luas tanah untuk perumahan besar (>25.000 sqft).
*   INDUS → Persentase area yang digunakan untuk bisnis non-ritel.
*   CHAS → Dummy variable (1 = berbatasan dengan sungai Charles, 0 = tidak).
*   NOX → Konsentrasi polusi udara (nitrogen oksida) dalam ppm.
*   RM → Rata-rata jumlah kamar per rumah (semakin tinggi, semakin mahal rumah).
*   AGE → Persentase rumah yang dibangun sebelum tahun 1940.
*   DIS → Jarak rata-rata ke pusat pekerjaan utama di Boston.
*   RAD → Indeks aksesibilitas ke jalan raya utama.
*   TAX → Tarif pajak properti per $10.000.
*   PTRATIO → Rasio jumlah murid terhadap guru di kota (semakin tinggi, semakin kurang optimal pendidikan).
*   B → Proporsi penduduk keturunan Afrika-Amerika (dihitung dengan rumus tertentu).
*   LSTAT → Persentase penduduk dengan status ekonomi rendah.
*   MEDV → Median nilai properti dalam ribuan dolar (sering dijadikan target prediksi).

